### PR TITLE
Value to tuple

### LINKED ISSUE #508: value_to tuple/pair
`value_to` should work for tuple-like and pair-like types. This interacts with #466 because `array` is tuple-like and needs to be handled as a special case.

### PR DESCRIPTION
**Pull request by grisumbras**
Fix #508

### CODE REVIEW

File: include/boost/json/detail/index_sequence.hpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# include <type_traits>
+#endif
+
+BOOST_JSON_NS_BEGIN
+namespace detail {
+
+#if ! defined(BOOST_JSON_STANDALONE)
+
+template <std::size_t... Is>
+using index_sequence = boost::mp11::index_sequence<Is...>;
```
Comment (vinniefalco) [inline include/boost/json/detail/index_sequence.hpp:26]: Very nice

**Comment by vinniefalco** [comment]
I really don't have the skills to review stuff related to `value_to` and `value_from`. @pdimov ?

**Comment by pdimov** [comment]
If it passes msvc-14.0 CI it's probably good.

**Comment by pdimov** [comment]
Looks good to me. To repeat what I said on Slack, now that tuple-likes are supported it looks like general maps (where the key is not string-like) would also start working (being treated as sequences of pairs.) If they do work, we need to have a test for it. If they don't, we might want to look into why.

**Pull request merged**